pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -230,6 +231,10 @@ type PersistentVolumeController struct {
 
 	createProvisionedPVRetryCount int
 	createProvisionedPVInterval   time.Duration
+
+	// For testing only: hook to intercept CSI driver name <=> Intree plugin name mapping
+	// Not used when set to nil
+	csiNameFromIntreeNameHook func(pluginName string) (string, error)
```

Comment:
In long term it would be better if CSI translation library provided an interface instead of function calls so its callers can create fake implementations for unit tests. As a separate PR, of course, no need to fix it here.

Reply:
Agree.

Reply:
xref" https://github.com/kubernetes/kubernetes/issues/74594